notifier/notifier.go

```diff
@@ -79,15 +88,16 @@ type Options struct {
 }
 
 type alertMetrics struct {
-	latency       *prometheus.SummaryVec
-	errors        *prometheus.CounterVec
-	sent          *prometheus.CounterVec
-	dropped       prometheus.Counter
-	queueLength   prometheus.GaugeFunc
-	queueCapacity prometheus.Gauge
+	latency                        *prometheus.SummaryVec
+	errors                         *prometheus.CounterVec
+	sent                           *prometheus.CounterVec
+	dropped                        prometheus.Counter
+	queueLength                    prometheus.GaugeFunc
+	queueCapacity                  prometheus.Gauge
+	alertmanagerDiscoveryCollector *alertmanagerDiscoveryCollector
 }
```

Comment:
If you look at `queueLength` there's `prometheus.NewGaugeFunc` being used, which sort of acts as a collector shortcut on a function.
One could skip the whole `Describe` and `Collect` hassle by just doing the same with a function such as:

```go
var n Notifier
// ...
f := func() float64 { return float64(len(n.Alertmanagers())) }
```

That function could just be passed to the metrics constructor.